src/content/app/tools/blast/state/action-listeners/blastActionListeners.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { PayloadAction, ListenerEffectAPI } from '@reduxjs/toolkit';
+
+import config from 'config';
+
+import { submitBlast } from '../blast-api/blastApiSlice';
+import {
+  restoreBlastSubmissions,
+  updateJob
+} from 'src/content/app/tools/blast/state/blast-results/blastResultsSlice';
+
+import {
+  saveBlastSubmission,
+  updateSavedBlastJob
+} from 'src/content/app/tools/blast/services/blastStorageService';
+
+import { POLLING_INTERVAL } from './blastActionListenerConstants';
+
+import type { BlastSubmission } from '../blast-results/blastResultsSlice';
+import type { RootState, AppDispatch } from 'src/store';
+
+export const submitBlastListener = {
+  matcher: submitBlast.matchFulfilled,
+  effect: async (
+    action: PayloadAction<{
+      submissionId: string;
+      submission: BlastSubmission;
+    }>,
+    listenerApi: ListenerEffectAPI<RootState, AppDispatch>
+  ) => {
+    const { submissionId, submission } = action.payload;
+    const { dispatch, signal } = listenerApi;
+
+    await saveBlastSubmission(submissionId, submission);
+
+    const runningJobs = submission.results.map(({ jobId }) => ({
+      submissionId,
+      jobId
+    }));
```

Comment:
@msmanoj here and below, updated the variable name from jobIds to runningJobs, according to your [comment](https://github.com/Ensembl/ensembl-client/pull/719#discussion_r837525133 ) from https://github.com/Ensembl/ensembl-client/pull/719.